src/resources/cpp/language-basics.md

```diff
@@ -1,15 +1,307 @@
+part. C++ provides integer types that can support different ranges of values.
+
+| Type          | Minimum Range                                           | Minimum Width in Bits |
+| ------------- | ------------------------------------------------------- | --------------------- |
+| `signed char` | [-128, 127]                                             | 8                     |
+| `short`       | [-32768, 32767]                                         | 16                    |
+| `int`         | [−32768, 32767]                                         | 16                    |
+| `long`        | [−2,147,483,648, 2,147,483,647]                         | 32                    |
+| `long long`   | [−9,223,372,036,854,775,808, 9,223,372,036,854,775,807] | 64                    |
+
+The above types are called signed types. This means that these types can represent negative values. C++ also has
+unsigned types, only supporting non-negative types.
+
+| Type                 | Minimum Range                   | Minimum Width in Bits |
+| -------------------- | ------------------------------- | --------------------- |
+| `unsigned char`      | [0, 255]                        | 8                     |
+| `unsigned short`     | [0, 65535]                      | 16                    |
+| `unsigned int`       | [0, 65535]                      | 16                    |
+| `unsigned long`      | [0, 4,294,967,295]              | 32                    |
+| `unsigned long long` | [0, 18,446,744,073,709,551,615] | 64                    |
+
+::: info
+
+These are just minimum ranges for values. Many platforms use different data models. On 64-bit Windows, the signed and
+unsigned variants of `int` is 32 bits in width. On 64-bit \*nix systems, a signed or unsigned `int` is 32 bits and a
+signed or unsigned `long` is 64 bits.
+
+:::
+
+### Character Types
+
+In addition to `signed char` and `unsigned char`, C++ has a type `char`. All three of these types are used for character
+representation. The standard does not specify if `char` is signed or unsigned, but its signedness is determined by the
+compiler and is usually the most optimal representation for a single byte character on the platform. On x86-64, char is
+typically signed, whereas on ARM64, it is typically unsigned.
+
+For characters that are not able to be represented in a single byte, C++ provides other distinct types for wider
+character storage.
+
+- `wchar_t` - A character type that is typically wider than the `char` type. On \*nix platforms, it is typically 32 bits
+  in width. On windows platforms, it is typically 16 bits in width.
+- `char8_t` - A character type that can represent any UTF-8 character representation.
+- `char16_t` - A character type that can represent any UTF-16 code point.
+- `char32_t` - A character type that can represent any UTF-32 code point.
+
+### Floating Point Types
```

Comment:
The more correct spelling, usually preferred in scientific settings is "floating-point", and it's spelled that way in the C++ standard. This is also strictly more correct because "floating-point" is a compound adjective, so the English grammar requires it to be hyphenated.